Only show Labels applicable to side

yoshnopa opened the pull request:
Like discussed in #377, I created a fork only showing specific labels for specific pages, like for example brunette may only apply to actors, but not to scenes, presuming scenes do not inherit them. 
If that is the case (or other scenarios regarding images or movies), the label will not show up.

I settled with this on option 2 of what was proposed in the issue, as this will take care of it automatically and it was not that difficult to implement. 

One thing before merging, I was able to get it working with the testserver, but not with the built release. Probably I simply forgot something, but the webinterface was not adjusted (the changes in the vue files). However, the server part is tested compiled and as a test setup, both times working. 

If there are further requirements, addings or thoughts, ask away!

boi123212321 commented on src/graphql/resolvers/query.ts:
```diff
@@ -122,6 +123,16 @@ export default {
     return labels.sort((a, b) => a.name.localeCompare(b.name));
   },
+  async getAreaLabels(_, { idfront }: { idfront: string }) {
+    const references = await CrossReference.getAll()
```
I have some performance concerns about the approach of this function.
The cross references table is the biggest table and can easily exceed 1 million documents. Doing a scan everytime to get some labels is pretty inefficient.
Alternatively, get all items of a table (scenes or images etc), and map those to their labels.
This would mean either 1 function per table, or another function parameter.

boi123212321 commented on src/graphql/resolvers/query.ts:
```diff
@@ -122,6 +123,16 @@ export default {
+        r => Label.getById(r.to)
+      )
+    ).filter(Boolean) as Label[];
+    return labels.sort((a, b) => a.name.localeCompare(b.name));
```
Plus the function is not checking for duplicates in the array, so the returned array is much bigger than it needs to be.
Take a look at the createObjectSet function in types/actor.

boi123212321 commented on src/graphql/schema/label.ts:
```diff
@@ -4,6 +4,7 @@ export default gql`
   extend type Query {
     numLabels: Int!
     getLabels: [Label!]!
+    getAreaLabels(idfront: String!): [Label]!
```
```suggestion
    getAreaLabels(idfront: String!): [Label!]!
```
We never want to have a null value in the array.

yoshnopa commented:
Thanks for your patience, I removed duplicates, corrected the schema and changed the system to search for the table first and extract all labels from there. 
However, this system isn't faster in my test enviroment of about 7000 references. Nevertheless, you're probably right that this will scale better.

boi123212321 commented:
in 0.20 the way label references are stored has changed.
there is now a separate table just for that, and it has an index on the type of item (scene, actor etc), which should be used.
just doesn't work for movies, they need to be done another way.
